src/lib/flatten/flatten.ts

```diff
@@ -57,8 +57,18 @@ export async function flattenToFesm15(opts: FlattenOpts): Promise<string> {
 
 export async function flattenToFesm5(opts: FlattenOpts): Promise<string> {
   const destFile = path.resolve(opts.outDir, 'esm5', opts.flatModuleFile + '.js');
+  // a temp file is required because of the sourcemaps remapping
+  const destTempFile = path.resolve(opts.outDir, 'esm5-downlevel-temp', opts.flatModuleFile + '.js');
 
-  await downlevelWithTsc(opts.entryFile, destFile);
+  await downlevelWithTsc(opts.entryFile, destTempFile);
+  // the below is required in order to avoid multiple imports for tslib
+  await rollupBundleFile({
```

Comment:
Hm, I'd prefer to do the down-levelling then in a rollup plugin

For example with bublé : https://github.com/rollup/rollup-plugin-buble

Or register the tsc downlevelling as a transform rollup plugin https://github.com/rollup/rollup/wiki/Plugins#creating-plugins:

```ts
rollup.plugins = [
   ...existinPlugins,
   {
      transform: (source, id ) => {
         const transpiled = ts.transpileModule(/* ... */);

         return {
           code: transpiled.outputText,
           map: transpiled.sourceMap
         };
      }
   }
]
```

Reply:
It's cleaner I agree

Reply:
Should I update your PR?

Reply:
I updated my PR with some code changes as suggested. However, If you need to do some more changes. Feel free.

Reply:
Oh, you already did!